client/src/components/ui/progress.tsx

```diff
@@ -0,0 +1,85 @@
+import Image from "next/image";
+import * as React from "react";
+
+export default function ProgressBar({ pageName }: { pageName: string }) {
+  const [progressWidth, setProgressWidth] = React.useState(0);
+  let progress = 0;
+  let titleName = "";
+
+  switch (pageName) {
+    case "schedule":
+      progress = 0;
+      titleName = "Drone Competition Schedule";
+      break;
+    case "format-rules":
+      progress = 30;
+      titleName = "Format & Rules";
+      break;
+    case "guests-sponsors":
+      titleName = "Guests & Sponsors";
+      progress = 60;
+      break;
+    case "leaderboard":
+      titleName = "Leaderboard";
+      progress = 80;
+      break;
+  }
```

Comment:
You're using `let progress = 0; let titleName = "";` and then mutating them via `switch`. This is fine, but less readable and error-prone than a simple map object approach. 

i.e.

```
const pageConfig: Record<string, { progress: number; title: string }> = {
  schedule: { progress: 0, title: "Drone Competition Schedule" },
  "format-rules": { progress: 30, title: "Format & Rules" },
  "guests-sponsors": { progress: 60, title: "Guests & Sponsors" },
  leaderboard: { progress: 80, title: "Leaderboard" },
};

...

const { progress, title } = pageConfig[pageName] ?? {
  progress: 0,
  title: "Untitled",
};
```